servatrice/src/serversocketinterface.cpp

```diff
@@ -946,14 +960,21 @@ Response::ResponseCode AbstractServerSocketInterface::cmdRegisterAccount(const C
         {
             QSqlQuery *query = sqlInterface->prepareQuery("insert into {prefix}_activation_emails (name) values(:name)");
             query->bindValue(":name", userName);
-            if (!sqlInterface->execSqlQuery(query))
+            if (!sqlInterface->execSqlQuery(query)) {
+                rc.setResponseExtension(re);
                 return Response::RespRegistrationFailed;
+            }
 
+            re->set_custom_reason_str(servatrice->getCustomRegSuccessMessage().toStdString());
+            rc.setResponseExtension(re);
```

Comment:
This call gets duplicated all over. Can we move it out so it's only in the source once? Maybe right after allocating the response object?